src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetInfoService.cs

```diff
@@ -123,64 +123,60 @@ private void PopulateSnippetCaches()
         {
             Debug.Assert(_expansionManager != null);
 
-            // Ideally we'd fork execution here based on whether the expansion manager is an
-            // IExpansionManager or not. Unfortunately, we cannot mention that type by name until
-            // the Roslyn build machines are upgraded to Visual Studio 2015 Update 1. We therefore
-            // need to try using IExpansionManager dynamically, from a background thread. If that
-            // fails, then we come back to the UI thread for using IVsExpansionManager instead.
-
             var token = _waiter.BeginAsyncOperation(GetType().Name + ".Start");
 
-            Task.Factory.StartNew(async () => await PopulateSnippetCacheOnBackgroundWithForegroundFallback().ConfigureAwait(false),
-                            CancellationToken.None,
-                            TaskCreationOptions.None,
-                            TaskScheduler.Default).CompletesAsyncOperation(token);
+            var asyncExpansionManager = _expansionManager as IExpansionManager;
```

Comment:
@davkean, this is to mainly support finding obscure bugs after we've released this to customers.  To unblock customers quickly, they can just revert back to the old implementation (using the old snippet manager native implementation), while we address the real product issue for the next release.  For example, we tried to release this for update 1, but found an obscure localization bug that caused many customers in non en-us locales to not have snippets.  To get around this, all they had to do was to add a regkey instead of waiting for us to publish a new update.